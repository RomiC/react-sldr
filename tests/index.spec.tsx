import React from 'react';
import renderer from 'react-test-renderer';

import Slider from '../src';

let props = null;
let slides = null;

beforeEach(() => {
  props = {
    className: 'custom-slider-class',
    controls: false,
    slidesAtOnce: 2
  };
  slides = [
    <img key="1" src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide1" />,
    <img key="2" src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide2" />,
    <img key="3" src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide3" />,
    <img key="4" src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide4" />,
    <img key="5" src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide5" />,
    <img key="6" src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide6" />
  ];
});

test('shoud render component', () => {
  const component = renderer.create(<Slider>{slides}</Slider>).toJSON();
  expect(component).toMatchSnapshot();
});

test('should render component with custom props', () => {
  const component = renderer.create(<Slider {...props}>{slides}</Slider>).toJSON();
  expect(component).toMatchSnapshot();
});