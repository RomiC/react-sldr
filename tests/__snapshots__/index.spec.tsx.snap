// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud render component 1`] = `
<div
  className="slider "
  onPointerCancel={[Function]}
  onPointerDown={[Function]}
  onPointerMove={[Function]}
  onPointerUp={[Function]}
>
  <div
    className="slider__wrapper"
    style={
      Object {
        "transform": "translate3d(0%,0,0)",
      }
    }
  >
    <ul
      className="slider__list"
    >
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide1"
        />
      </li>
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide2"
        />
      </li>
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide3"
        />
      </li>
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide4"
        />
      </li>
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide5"
        />
      </li>
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide6"
        />
      </li>
    </ul>
  </div>
  <div
    className="slider__controls"
  >
    <button
      className="slider__control slider__control--prev"
      onClick={[Function]}
      type="button"
    >
      ←
    </button>
     
    <button
      className="slider__control slider__control--next"
      onClick={[Function]}
      type="button"
    >
      →
    </button>
  </div>
</div>
`;

exports[`should render component with custom props 1`] = `
<div
  className="slider custom-slider-class"
  onPointerCancel={[Function]}
  onPointerDown={[Function]}
  onPointerMove={[Function]}
  onPointerUp={[Function]}
>
  <div
    className="slider__wrapper"
    style={
      Object {
        "transform": "translate3d(0%,0,0)",
      }
    }
  >
    <ul
      className="slider__list"
    >
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide1"
        />
      </li>
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide2"
        />
      </li>
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide3"
        />
      </li>
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide4"
        />
      </li>
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide5"
        />
      </li>
      <li
        className="slider__item"
        draggable={false}
      >
        <img
          src="http://placehold.it/1000x400/ffffff/c0392b/&text=slide6"
        />
      </li>
    </ul>
  </div>
</div>
`;
